src/contracts/pods/EigenPod.sol

```diff
@@ -280,6 +289,41 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
         }
     }
 
+    function submitPartialWithdrawalsBatchForVerification(
+        uint64 oracleTimestamp,
+        uint64 startTimestamp,
+        uint64 endTimestamp,
+        bytes32 FUNCTION_ID
+    ) external {
+        eigenPodManager.requestProofViaFunctionGateway(
+            FUNCTION_ID, 
+            startTimestamp,
+            endTimestamp,
+            address(this),
+            oracleTimestamp,
+            requestNonce,
+            this.handleCallback.selector
+        );
+        /**
+        * Zero out the running total of partial withdrawals claimed.  Once another merkle proof for partial withdrawals is proven
+        * this running total wil once again start accumulating.
+        */
+        sumOfPartialWithdrawalsClaimedGwei = 0;
+        requestNonce++;
+
+        emit PartialWithdrawalProofRequested(startTimestamp, endTimestamp, requestNonce);
+    }
+
+    /// @notice The callback function for the ZK proof fulfiller.
+    function handleCallback(bytes memory output, bytes memory context) external onlyFunctionGateway {
+        uint256 partialWithdrawalSumWei;
+        assembly {
+            partialWithdrawalSumWei := mload(add(output, 0x20))
+        }
+        uint256 requestNonce = abi.decode(context, (uint256));
+        emit PartialWithdrawalProven(requestNonce, partialWithdrawalSumWei);
+    }
```

Comment:
so after here is when we would calculate the amount of ETH to send + actually forward it on